src/EventFactory.php

```diff
@@ -61,15 +61,18 @@ public function __construct(SerializerInterface $serializer, RepresentationSeria
      */
     public function createWithStacktrace(array $payload): Event
     {
-        $event = $this->create($payload);
-
-        if (!$event->getStacktrace()) {
-            $stacktrace = Stacktrace::createFromBacktrace($this->options, $this->serializer, $this->representationSerializer, debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS), __FILE__, __LINE__);
-
-            $event->setStacktrace($stacktrace);
+        if (!isset($payload['stacktrace']) || !$payload['stacktrace'] instanceof Stacktrace) {
+            $payload['stacktrace'] = Stacktrace::createFromBacktrace(
+                $this->options,
+                $this->serializer,
+                $this->representationSerializer,
+                debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS),
+                __FILE__,
+                __LINE__ - 6
```

Comment:
I would create this value as `__LINE__ + 1` one line before this call, to make it more robust and less change-prone while refactoring.

Reply:
Actually it doesn't really matter because:

- If we need to add one more argument to the method we still have to update this line taking into account the number of new lines
- It's safer to rely on the `-` because we cannot remove any argument to avoid BC, the only thing that can change is that we revert the formatting of this code to one-line again which is unlikely to happen at this point
- It's more appropriate to point to the line where the function name actually is rather than on its closing parenthesis

Reply:
I was not suggesting to change it inline, rather to create a temp variable before the call, and reference the call itself with the `+1`. This way we no longer have to worry about any of the concerns that you pointed out.

Reply:
>This way we no longer have to worry about any of the concerns that you pointed out

It could very well be that I'm not understanding your concerns, but I cannot see how your proposal would solve the issue: even creating a temporary variable (which I would like to avoid as it's pretty useless) right before the function call, if someone adds a new line between them the sum will not be valid regardless. In the end, it will always matter how many lines there are between __LINE__ and the line where you call the function

Reply:
Sure. But putting a line in between those two calls is calling for trouble and you would have to do it intentionally. Instead, with this situation, this could break up with automatic CS fixes too.

BTW, I would be fine with either if we have a unit test that checks this, it's an important information to get right.

Reply:
>But putting a line in between those two calls is calling for trouble and you would have to do it intentionally

Also adding a new argument or line before this or removing one argument must be done intentionally, so either way we always go back to the same conclusion

>I would be fine with either if we have a unit test that checks this

I don't think it's possible to test this as it would mean hardcode the line number into the test itself, which is something I'm pretty sure neither of us wants. We never bothered before about this kind of things, so I strongly think that it's worthless continuing the discussion